subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -578,6 +591,11 @@ static void timeout(struct k_work *work)
 		CONTAINER_OF(work, struct bt_mesh_light_ctrl_srv, timer.work);
 
 	if (!is_enabled(srv)) {
+		if (CONFIG_BT_MESH_LIGHT_CTRL_SRV_RESUME_DELAY &&
+		    atomic_test_and_clear_bit(&srv->flags, FLAG_RESUME_TIMER)) {
```

Comment:
Does this flag have a value. If the srv is disabled this flag is set.  There is also a possibility that this flag can be left set if after a timeout if the srv is enabled. It is not necessary  any problem with that.

Reply:
The idea is to prevent misfires of the timeout if the timer cancel call in `ctrl_disable` fails. This cannot happen when we immediately schedule the timer again after disable, but if the cancel call in model_reset fails, the model will come back from the dead when the timer fires.

You're right about the flag not being cleared properly. Good catch, I'll add a line to clear it in `ctrl_enable`, and will double check that it doesn't happen anywhere else